protocol/signalfx/trace_test.go

```diff
@@ -0,0 +1,121 @@
+package signalfx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/signalfx/golib/trace"
+)
+
+type fakeSink struct {
+	handler func([]*trace.Span)
+}
+
+func (s *fakeSink) AddSpans(ctx context.Context, spans []*trace.Span) error {
+	s.handler(spans)
+	return nil
+}
+
+// Benchmark the case where all spans are Zipkin v2.
+func BenchmarkTraceDecoder(b *testing.B) {
+	reqBody := `[
```

Comment:
might be better to have a single trace, and put it say 100 times.  2 traces, is not going to exhibit the same behavior as i was referring to i don't think..